SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DeviceInformationHelper.cs

```diff
@@ -123,6 +126,38 @@ protected override string GetScreenSize()
             }
         }
 
+        private static string DeploymentVersion
+        {
+            get
+            {
+#if NET461
+                // Get ClickOnce version (does not exist on .NET Core). 
+                if (ApplicationDeployment.IsNetworkDeployed)
+                {
+                    return ApplicationDeployment.CurrentDeployment.CurrentVersion.ToString();
+                }
+#endif
+                return null;
+            }
+        }
+
+        private static string FileVersion
+        {
+            get
+            {
+                // The AssemblyFileVersion uniquely identifys a build.
+                var entryAssembly = Assembly.GetEntryAssembly();
+                if (entryAssembly != null)
+                {
+                    var fileVersionInfo = FileVersionInfo.GetVersionInfo(entryAssembly.Location);
+                    return fileVersionInfo.FileVersion;
+                }
+
+                // Fallback if entry assembly is not found (in unit tests for example).
+                return Application.ProductVersion;
```

Comment:
But the way the code is written, it would make more sense to return null here and use ?? in the caller

Reply:
Initially I thought that it makes more sense, but if take into account why it's here and what this fallback really is, then it's better to keep it as is.
This line is mainly for unit tests (or any unknown cases), where entry assembly is not available. It will be just more confusing if move this fallback to caller since it's not supposed to be called in regular situation. The line already has a comment about it.

Let's keep this fallback here to remain code in main function clear.